app/views/nic/_base_form.html.erb

```diff
@@ -5,8 +5,9 @@
 
 <%= text_f f, :mac,
               :label => _("MAC Address"),
-              :label_help => _("Media access control address for this interface. Format must be six groups of two hexadecimal digits <br/> separated by colons (:), e.g. 00:11:22:33:44:55. Most virtualization compute resources (e.g. VMware, oVirt, libvirt) <br/> will overwrite the provided value with new random MAC address.").html_safe,
+              :label_help => _("Media access control address for this interface. Format must be six groups of two hexadecimal digits <br/> separated by colons (:), e.g. 00:11:22:33:44:55. Most virtualization compute resources (e.g. VMware, oVirt, libvirt) <br/> will provide new random MAC address.").html_safe,
               :label_help_options => { :rel => 'popover-modal' },
+              :disabled => f.object.host&.compute_resource&.provide_attribute?(:mac),
```

Comment:
Oh, it doesn't work for unmanaged hosts :thinking:
We will have to use `try` as `Host::Base` doesn't have `:compute_resource` relation defined.
```suggestion
              :disabled => f.object.host.try(:compute_resource)&.provide_attribute?(:mac),
```

Reply:
Just to clarify, `Host::Base` is an abstract class and there should never be an instance of it. It's common parent for `Host::Managed` and `Host::Discovered` classes (the later is defined in foreman_discovery plugin). We shouldn't need to worry about it.

Unmanaged hosts are hosts which are not provisioned from Foreman, or more precisely, we don't manage their information in external systems such as DHCP/DNS/TFTP/Realm... They are still instances of `Host::Managed` however the `managed` column in SQL is set to `false`. So instances of `Host::Managed` can be either managed or unmanaged based on the model attributes.

Yes, the name of a class is confusing and we should perhaps add `abstract` to `Host::Base` to make it clearer.

Reply:
This is all true, thanks Marek for clafifying. We have an test failures because of this though, we use `Host::Base` there in `hosts_controller_test`, what is also wrong (even rendering templates there is probably wrong).
I would argue, we should keep the partial renderable with only `Host::Base` model (as we should be able to inherit from the base class with minimal additions, where `compute_resource` association don't need to be part of it) and still be able to render the basic form for it. As this require us to make only my suggested change (see above) I think it's worth it :)

Reply:
Have you considered using this method?

https://github.com/theforeman/foreman/blob/d5dc07f3e63d63cd43be97f1cefe452ff4437587/app/models/concerns/orchestration/compute.rb#L29

Reply:
I'd even say we should merge Host::Base with Host::Managed and Host::Discover on the first occasion. But Timo's suggestion would probably do in this case, so plus one to that

Reply:
Thanks Timo for the suggestion :+1: . We still need to use `try` as `Orchestration::Compute` is not included in the `Host::Base` :(

@domitea but like this, it would work, right?

```suggestion
              :disabled => f.object.host.try(:compute_provides?, :mac),
```
P.S. I guess I'd be ok fixing the test by changing class in respective places to `Host::Managed` as I agree with Marek here, that `Host::Base` is not really usefull.

Reply:
> P.S. I guess I'd be ok fixing the test by changing class in respective places to Host::Managed as I agree with Marek here, that Host::Base is not really usefull.

as long as the test works, I guess it may have side effects, perhaps with `:managed => false` would work fine

Reply:
> as long as the test works, I guess it may have side effects, perhaps with `:managed => false` would work fine

I don't get what you mean, but we're failing on https://github.com/theforeman/foreman/blob/c95dc3473105aa523c11cde777bdcd85961c9c6e/test/controllers/hosts_controller_test.rb#L1568 what should probably not render the templates and it can be easily switched to `Host::Managed` but I would rather have it working for `Host::Base` if it's not much of a pain and I don't think it is it's only `try` over `&.`

Reply:
ok, sorry I was a bit brief from the phone...

I think we should not be using `Host::Base` in our tests, because it's not real scenario. We should be testing the real objects that user will use, at least in controller test.

I assume the reason for using `Host::Base` in that test was, it's much easier to create such instance. The `Host::Managed` object can cause more side effects, e.g. could trigger the provisioning orchestration when we call `.save` on it. This can be avoided by setting `managed: false`. It may not be needed at all and perhaps all would work if we just switch `Host::Base` to `Host::Managed` in the test, which is IMHO the best solution.

But I'd also accept using `try` or or whatever else solution suggested here.